asonix/models

asonix opened the pull request:
Add proper models support.

I'd like some help figuring out the best way to integrate the old migrations (in the root directory) with the new migrations (in the aardwolf-models directory).

The reason I haven't integrated them yet is because I'm not sure I'm comfortable creating the database with a hardcoded name, and I'm also not sure I want the user running the migrations to have permission to add pgcrypto and create databases and such.

It's better, in my opinion, to create the db as an admin with the required modules, and then let a less-permissioned user be used by the application.

As far as how integrated aardwolf-models is with aardwolf-server, it's at parity with the previous implementation.

jfmcbrayer commented:
I'm a little late on this,  and am going to be in meetings most of the day. Will try to review at lunch, I hope. 

I share concerns about the privileges needed to add the module, not necessarily about database creation. Agree about hard coded database name.

The old migrations don't need to stay around, IMO. They only provided logins, and this code should also do that.

BanjoFox commented on aardwolf-models/examples/local_user_auth.rs:
```diff
@@ -0,0 +1,210 @@
+        Ok(())
+    });
+
+    println!("Hewwo, Mr Obama???");
```
Misspelling on 209

Should be "Thanks Obama!" 

;D

asonix replied:
>:c

asonix replied:
actually though, I need to update that example

BanjoFox replied:
*pats* its a joke fren

asonix commented:
I updated the auth controller to do a bit better error handling. I've also updated the example local_user_auth.rs to use internal methods I created for the auth controller.

As far as what to do about the migrations, I'd really like a comment from @pwoolcoc

asonix commented:
@BanjoFox I've already deleted the User and Account migrations. I'm specifically asking about the migrations that create the database, add the pgcrypto, and set the search_path

BanjoFox commented:
Looks good to me then :)

asonix commented:
I'd like to handle #113 before merging this one
